Disable hex4 and hex8

elvisvoer opened the pull request:
Closes #432

casesandberg commented on src/helpers/spec.js:
```diff
@@ -92,15 +92,15 @@ describe('helpers/color', () => {
       expect(color.isValidHex('f')).toBeFalsy()
       expect(color.isValidHex('ff')).toBeFalsy()
       expect(color.isValidHex('fff')).toBeTruthy()
       // expect(color.isValidHex('ffff')).toBeFalsy()
```
Hey, you mind commenting this back in so that we have a guard for hex4 as well?

elvisvoer commented:
@casesandberg great catch. I realised I was only taking colors with leading hash even though the test I modified was without. Fixed both implementation and uncommented assertion.

casesandberg commented:
Wonderful @elvisvoer, thanks for your work on this!

casesandberg commented:
I've invited you to join the repo as a collaborator – no pressure to accept! Feel free to reach out with any questions.

elvisvoer commented:
@casesandberg thanks for merging and for the invitation. I will reach out to you with questions for sure.

casesandberg commented:
This is live in `react-color@2.14.0` 🎉

leewz commented on src/helpers/color.js:
```diff
@@ -46,7 +46,9 @@ export default {
   isValidHex(hex) {
-    return tinycolor(hex).isValid()
+    // disable hex4 and hex8
+    const lh = (String(hex).charAt(0) === '#') ? 1 : 0
```
Minor point, but I'm curious: why isn't this `hex[0] == '#'`? Since length is checked, `hex` is expected to be a string.

Another alternative:
```js
const len = hex.startsWith('#') ? hex.length - 1 : hex.length
return (len == 3 || len == 6) && tinycolor(hex).isValid()
```

By checking that length is 3 or 6, instead of NOT 4 or >6 (per the spec of this function), it won't matter if tinycolor later allows other lengths.

UnHumbleBen commented:
> This is live in `react-color@2.14.0`

Can this be fixed in version 3.0.0 as well? I would like UMD bundle with this fix, and version 3 currently supports UMD. Otherwise, please make UMD bundle for version 2
